test/integration/redirects_test.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require("test_helper")
+
+class RedirectsTest < IntegrationTestCase
+  def test_controller
+    get("/article")
+    assert_equal("/articles", @response.request.fullpath)
+  end
+
+  def test_list_article
+    get("/article/list_article")
+    assert_equal("/articles", @response.request.fullpath)
+  end
+
+  def test_index_article
+    get("/article/index_article/")
+    assert_equal("/articles", @response.request.fullpath)
+  end
+
+  def test_show_article
+    get("/article/show_article/#{Article.first.id}")
+    assert_equal("/articles/#{Article.first.id}", @response.request.fullpath)
+  end
+
+  def test_create_article_get
+    login!("article_writer", "testpassword", true)
```

Comment:
I don't see a way to avoid `login!` 
Maybe @mo-nathan can pull another rabbit out of his hat.
I could avoid multiple calls to `login!` by combining all the methods that call it.